classes.py

```diff
@@ -28,9 +34,28 @@ class ChangeSet():
     """Pages that were added in the diff"""
     removed: List[Page]
     """Pages that were removed in the diff"""
-    changed: List[Tuple[Page, List[PropertyChange]]]
+    changed: List[Tuple[Page, List[PropertyChange], datetime]]
     """Pages that had their properties changed in the diff. The key is the page id and the value is the list of 
     properties that were changed"""
 
     def is_empty(self) -> bool:
         return len(self.added) == 0 and len(self.removed) == 0 and len(self.changed) == 0
+
+    def merge_set(self, other_changed_set):
+        """merges the changes from the other set into the current set
+        """
+
+        # For the added and removed pages, we simply add them to the new set if there is any
+        self.added += other_changed_set.added
+        self.removed += other_changed_set.removed
+        
+
+        # We compare if there are any changes in the same page and coalece
+        # Horribly underoptimized, we will get back to this later
```

Comment:
I agree. If you want to do this you should probably use a different data structure e.g. a dict that maps pages (or page IDs) to their changes

Reply:
suggestion: refactor `Page` into a dataclass so you can implement `__hash__` and then change `ChangeSet.changed` into a `Dict[Page, Tuple[List[PropertyChange], datetime]]` or something